plugins/dfu/dfu-tool.c

```diff
@@ -2184,26 +2184,27 @@ main (int argc, char *argv[])
 	g_autoptr(DfuToolPrivate) priv = g_new0 (DfuToolPrivate, 1);
 	g_autoptr(GError) error = NULL;
 	g_autoptr(GOptionContext) context = NULL;
+
+	setlocale (LC_ALL, "");
+
+	bindtextdomain (GETTEXT_PACKAGE, LOCALEDIR);
+	bind_textdomain_codeset (GETTEXT_PACKAGE, "UTF-8");
+	textdomain (GETTEXT_PACKAGE);
+
```

Comment:
ohh, how come this bit had to move? Do the GOptionEntry translated options not work otherwise?

Reply:
It works without the move but I suppose it's thanks some compiler optimizations and not general ok. If code would run with written sequence, GOptionEntry calls gettext() without gettext initialization.Hower I'm not an advanced hacker and may be it's my bad assumption.

Reply:
@Marv-CZ unfortunately this approach won't work.  It's against ISO C90.

```
error: ISO C90 forbids mixed declarations and code [-Werror=declaration-after-statement]
```

I guess i'd ask - does this work with `fwupdmgr` (`fu-util.c`) or `fwuptdool` (`fu-tool.c`)?  The approach used for `dfu-tool`, `fwupdmgr`, and `fwupdtool` should all be done the same I think.